### DISCUSSION TITLE
Should the .gitignore template for new gems include Gemfile.lock?

### DISCUSSION THREAD

**Discussion by alexandergitter** [ruby/rubygems#8317] [Q&A] [answered]
I just read https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/, which recommends not checking in Gemfile.lock into version control when developing a gem.
I checked some of the most downloaded gems, including aws, rake, rspec, json, i18n, minitest and others and sure enough, they all ignore Gemfile.lock. With Rails being a prominent exception.

Now I wonder - why does the default .gitignore template for new gems not list Gemfile.lock? Should it?

**Comment by deivid-rodriguez** [comment, marked as answer]
See https://github.com/rubygems/rubygems/issues/3372#issue-581057305. Indeed we initially followed that recommendation but that eventually changed.

**Reply by alexandergitter** [reply]
Makes sense, thank you for the quick response!

**Reply by deivid-rodriguez** [reply]
No problem!
